Sprint-3/todo-list/script.js

```diff
@@ -1,13 +1,51 @@
 function populateTodoList(todos) {
   let list = document.getElementById("todo-list");
   // Write your code to create todo list elements with completed and delete buttons here, all todos should display inside the "todo-list" element.
+  list.innerHTML = "";
+
+  todos.forEach((todo) => {
+    let li = document.createElement("li");
+    li.className = "list-group-item d-flex justify-content-between align-items-center";
+    
+    let textContent = todo.task;
+    if (todo.deadline) {
+      const deadline = new Date(todo.deadline);
+      const today = new Date();
+      const diffDays = Math.ceil((deadline - today) / (1000 * 60 * 60 * 24));
+      const countdownText = diffDays >= 0 ? `(${diffDays} days left)` : `(Overdue)`;
+      textContent += ` - Due: ${todo.deadline} ${countdownText}`;
+    }
+    li.textContent = textContent;
+
+    let badge = document.createElement("span");
+    badge.className = "badge bg-primary rounded-pill";
+
+    let checkIcon = document.createElement("i");
+    checkIcon.className = "fa fa-check";
+    checkIcon.style.cursor = "pointer";
+    checkIcon.addEventListener("click", () => {
+      li.style.textDecoration = li.style.textDecoration === "line-through" ? "" : "line-through";
```

Comment:
The `todos` array represent the state of the app, whereas the HTML + CSS represent the "view" of the app.
It is better to organise code in the following manner:
- Any change to the app (add new todo, delete completed todos, mark a todo as completed) should only change the states. 
- Then update the view based only on states.

This way, If the app were to save the todos in a database or in `window.localStorage`, the app can just save the state data (without saving any HTML or CSS).

So the code inside this `checkIcon`'s event listener should only update the `completed` property of the current todo, then update the view.
The code to conditionally apply a line-through should be placed somewhere around line 18.

Similar approach applies also to `trashIcon`'s event listener.